src/libsyntax/feature_gate.rs

```diff
@@ -1774,11 +1774,14 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
 
     fn visit_path(&mut self, path: &'a ast::Path, _id: NodeId) {
         for segment in &path.segments {
+            // Context of ident spans cannot be overriden to ignore unstable features,
+            // so replace it with path span context.
```

Comment:
I'm not sure what's happening here. Do you have an example where the behavior changes?

Reply:
It's a fix for the regression caught by Travis and mentioned in https://github.com/rust-lang/rust/pull/49154#issuecomment-374319190.

The problem is that here we want two different `SyntaxContext`s for identifiers generated by `#[test]` - one context for name resolution (unhygienic) and another one for stability checking (hygienic, stability is not checked), but `Ident` has only one `SyntaxContext`.

So we can keep the stability checking context in something that is not used for name resolution, for example whole `Path` (or anything else that is not an identifier/lifetime).

Reply:
Oh I see. Can the source comment be expanded to say more of what is happening from its own perspective? e.g. starting with "Here we check the span from the whole Path instead of that of individual Idents *specifically* because only the former can be ...".